examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/StringTests.elm

```diff
@@ -311,3 +311,172 @@ stringTrimLeft str =
 stringTrimRight : String -> String
 stringTrimRight str =
     String.trimRight str
+
+
+{-|
+
+    Test: String/fromChar
+    expected = "a"
+
+-}
+stringFromCharTest : TestContext -> String
+stringFromCharTest ctx =
+    test ctx
+        (String.fromChar 'a')
+
+
+{-|
+
+    Test: String/cons
+    expected = "abc"
+
+-}
+stringConsTest : TestContext -> String
+stringConsTest ctx =
+    test ctx
+        (String.cons 'a' "bc")
+
+
+{-|
+
+    Test: String/uncons
+    expected = Just ('a', "bc")
+
+-}
+stringUnconsTest : TestContext -> Maybe ( Char, String )
+stringUnconsTest ctx =
+    test ctx
+        (String.uncons "abc")
+
+
+{-|
+
+    Test: String/toList
+    expected = ['a', 'b', 'c']
+
+-}
+stringToListTest : TestContext -> List Char
+stringToListTest ctx =
+    test ctx
+        (String.toList "abc")
+
+
+{-|
+
+    Test: String/fromList
+    expected = "abc"
+
+-}
+stringFromListTest : TestContext -> String
+stringFromListTest ctx =
+    test ctx
+        (String.fromList [ 'a', 'b', 'c' ])
+
+
+{-|
+
+    Test: String/pad
+    pad 5 ' ' "1" == " 1 "
+    pad 5 ' ' "11" == " 11 "
+    pad 5 ' ' "121" == " 121 "
+    pad 5 ' ' "1234" == " 1234"
+    pad 5 ' ' "12345" == "12345"
+    pad 5 ' ' "123456" == "123456"
+    pad 0 ' ' "123" == "123"
+    pad -5 ' ' "123" == "123"
+    pad 5 ' ' "" == "     "
+
+-}
+stringPadTest : Int -> String -> String
+stringPadTest size input =
+    String.pad size ' ' input
+
+
+{-|
+
+    Test: String/map
+    expected = "a.b.c"
+
+-}
+stringMapTest : TestContext -> String
+stringMapTest ctx =
+    test ctx
+        String.map
+        (\c ->
+            if c == '/' then
+                '.'
+
+            else
+                c
+        )
+        "a/b/c"
+
+
+{-|
+
+    Test: String/filter
+    expected = "bc"
+
+-}
+stringFilterTest : TestContext -> String
+stringFilterTest ctx =
+    test ctx
+        (String.filter (\c -> c /= 'a') "abc")
+
+
+{-|
+
+    Test: String/foldl
+    input = UPPERCASE, expected = True
+    input = lowercase, expected = False
+    input = camelCase, expected = False
+
+-}
+stringFoldlTest : String -> Bool
+stringFoldlTest input =
+    String.foldl (\char acc -> acc && (char >= 'A' && char <= 'Z')) True input
+
+
+{-|
+
+    Test: String/foldr
+    input = "Hello, World", expected = 2
+    input = "HELLO, WORLD", expected = 10
+
+-}
+stringFoldrTest : String -> Int
+stringFoldrTest input =
+    String.foldr
+        (\char count ->
+            if char >= 'A' && char <= 'Z' then
+                count + 1
+
+            else
+                count
+        )
+        0
+        input
```

Comment:
We could just the examples from https://package.elm-lang.org/packages/elm/core/latest/String as the test cases, so that the difference between ```foldl``` and ```folds``` is clear?

```
foldl cons "" "time" == "emit"

foldr cons "" "time" == "time"
```

Reply:
done